### ISSUE TITLE
Again Closed socket 19 (AF_UNIX)

### ISSUE THREAD

**Issue by DS455** [neutrinolabs/xrdp#1116]
I used all the fixes from this forum, and could not connect.
xserver-xorg-dev removed.

Xubuntu 16.04 LTS
```
lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.4 LTS
Release:	16.04
Codename:	xenial
```

```
xrdp --version

xrdp: A Remote Desktop Protocol server.
Copyright (C) Jay Sorg 2004-2014
See http://www.xrdp.org for more information.
Version 0.9.6
```
Xwrapper.config
```
# Xwrapper.config (Debian X Window System server wrapper configuration file)
#
# This file was generated by the post-installation script of the
# xserver-xorg-legacy package using values from the debconf database.
#
# See the Xwrapper.config(5) manual page for more information.
#
# This file is automatically updated on upgrades of the xserver-xorg-legacy
# package *only* if it has not been modified since the last upgrade of that
# package.
#
# If you have edited this file but would like it to be automatically updated
# again, run the following command as root:
#   dpkg-reconfigure xserver-xorg-legacy
allowed_users=anybody

```
xrdp.log 
```
[20180428-14:49:27] [INFO ] starting xrdp with pid 806
[20180428-14:49:27] [INFO ] listening to port 3389 on 0.0.0.0
[20180428-14:52:01] [INFO ] Socket 12: AF_INET connection received from 192.168.0.27 port 44376
[20180428-14:52:01] [DEBUG] Closed socket 12 (AF_INET 192.168.0.66:3389)
[20180428-14:52:01] [DEBUG] Closed socket 11 (AF_INET 0.0.0.0:3389)
[20180428-14:52:01] [INFO ] Using default X.509 certificate: /etc/xrdp/cert.pem
[20180428-14:52:01] [INFO ] Using default X.509 key file: /etc/xrdp/key.pem
[20180428-14:52:02] [DEBUG] TLSv1.2 enabled
[20180428-14:52:02] [DEBUG] TLSv1.1 enabled
[20180428-14:52:02] [DEBUG] TLSv1 enabled
[20180428-14:52:02] [DEBUG] Security layer: requested 3, selected 1
[20180428-14:52:02] [INFO ] connected client computer name: r2d2-desktop
[20180428-14:52:02] [INFO ] TLS connection established from 192.168.0.27 port 44376: TLSv1 with cipher ECDHE-RSA-AES256-SHA
[20180428-14:52:02] [DEBUG] xrdp_00000448_wm_login_mode_event_00000001
[20180428-14:52:02] [INFO ] Loading keymap file /etc/xrdp/km-00000409.ini
[20180428-14:52:02] [WARN ] local keymap file for 0x00000409 found and doesn't match built in keymap, using local keymap file
[20180428-14:52:07] [DEBUG] xrdp_wm_log_msg: connecting to sesman ip 127.0.0.1 port 3350
[20180428-14:52:07] [INFO ] xrdp_wm_log_msg: sesman connect ok
[20180428-14:52:08] [DEBUG] xrdp_wm_log_msg: sending login info to session manager, please wait...
[20180428-14:52:08] [DEBUG] return value from xrdp_mm_connect 0
[20180428-14:52:08] [INFO ] xrdp_wm_log_msg: login successful for display 10
[20180428-14:52:08] [DEBUG] xrdp_wm_log_msg: started connecting
[20180428-14:52:11] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:52:15] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:52:19] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:52:22] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:52:26] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:52:29] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:52:33] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:52:36] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:52:40] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:52:44] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:52:47] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:52:51] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:52:54] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:52:58] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:53:02] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:53:05] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:53:09] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:53:12] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:53:16] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:53:19] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:53:23] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:53:26] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:53:30] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:53:34] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:53:37] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:53:41] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:53:44] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:53:48] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:53:51] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:53:55] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:53:59] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:54:02] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:54:06] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:54:09] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:54:13] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:54:17] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:54:20] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:54:24] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:55:42] [DEBUG] xrdp_wm_log_msg: connection problem, giving up
[20180428-14:55:42] [DEBUG] Closed socket 19 (AF_UNIX)
[20180428-14:55:42] [DEBUG] xrdp_wm_log_msg: some problem
[20180428-14:55:42] [DEBUG] xrdp_mm_module_cleanup
[20180428-14:55:42] [DEBUG] Closed socket 18 (AF_INET 127.0.0.1:54256)

```
xrdp-sesman.log
```
[20180428-14:52:07] [INFO ] A connection received from 127.0.0.1 port 54256
[20180428-14:52:08] [INFO ] ++ created session (access granted): username r2d2, ip 192.168.0.27:44376 - socket: 12
[20180428-14:52:08] [INFO ] starting Xorg session...
[20180428-14:52:08] [DEBUG] Closed socket 9 (AF_INET 0.0.0.0:5910)
[20180428-14:52:08] [DEBUG] Closed socket 9 (AF_INET 0.0.0.0:6010)
[20180428-14:52:08] [DEBUG] Closed socket 9 (AF_INET 0.0.0.0:6210)
[20180428-14:52:08] [DEBUG] Closed socket 8 (AF_INET 127.0.0.1:3350)
[20180428-14:52:08] [INFO ] calling auth_start_session from pid 1097
[20180428-14:52:08] [DEBUG] Closed socket 7 (AF_INET 127.0.0.1:3350)
[20180428-14:52:08] [DEBUG] Closed socket 8 (AF_INET 127.0.0.1:3350)
[20180428-14:52:08] [INFO ] Xorg :10 -auth .Xauthority -config xrdp/xorg.conf -noreset -nolisten tcp -logfile .xorgxrdp.%s.log
[20180428-14:52:18] [ERROR] X server for display 10 startup timeout
[20180428-14:52:18] [CORE ] waiting for window manager (pid 1102) to exit
[20180428-14:52:18] [ERROR] X server for display 10 startup timeout
[20180428-14:52:18] [ERROR] another Xserver might already be active on display 10 - see log
[20180428-14:52:18] [DEBUG] aborting connection...
[20180428-14:52:18] [CORE ] window manager (pid 1102) did exit, cleaning up session
[20180428-14:52:18] [INFO ] calling auth_stop_session and auth_end from pid 1097
[20180428-14:52:18] [DEBUG] cleanup_sockets:
[20180428-14:52:18] [DEBUG] cleanup_sockets: deleting /tmp/.xrdp/xrdp_chansrv_socket_10
[20180428-14:52:19] [DEBUG] cleanup_sockets: deleting /tmp/.xrdp/xrdpapi_10
[20180428-14:52:19] [DEBUG] cleanup_sockets: failed to delete /tmp/.xrdp/xrdpapi_10
[20180428-14:52:19] [INFO ] ++ terminated session:  username r2d2, display :10.0, session_pid 1097, ip 192.168.0.27:44376 - socket: 12
```
sesman.ini
```
[Globals]
ListenAddress=127.0.0.1
ListenPort=3350
EnableUserWindowManager=true
UserWindowManager=startwm.sh
DefaultWindowManager=startwm.sh

[Security]
AllowRootLogin=true
MaxLoginRetry=4
TerminalServerUsers=tsusers
TerminalServerAdmins=tsadmins
; When AlwaysGroupCheck=false access will be permitted
; if the group TerminalServerUsers is not defined.
AlwaysGroupCheck=false

[Sessions]
;; X11DisplayOffset - x11 display number offset
; Type: integer
; Default: 10
X11DisplayOffset=10

;; MaxSessions - maximum number of connections to an xrdp server
; Type: integer
; Default: 0
MaxSessions=50

;; KillDisconnected - kill disconnected sessions
; Type: boolean
; Default: false
; if 1, true, or yes, kill session after 60 seconds
KillDisconnected=false

;; DisconnectedTimeLimit - when to kill idle sessions
; Type: integer
; Default: 0
; if not zero, the seconds before a disconnected session is killed
; min 60 seconds
DisconnectedTimeLimit=0

;; Policy - session allocation policy
; Type: enum [ "Default" | "UBD" | "UBI" | "UBC" | "UBDI" | "UBDC" ]
; Default: Xrdp:<User,BitPerPixel> and Xvnc:<User,BitPerPixel,DisplaySize>
; "UBD" session per <User,BitPerPixel,DisplaySize>
; "UBI" session per <User,BitPerPixel,IPAddr>
; "UBC" session per <User,BitPerPixel,Connection>
; "UBDI" session per <User,BitPerPixel,DisplaySize,IPAddr>
; "UBDC" session per <User,BitPerPixel,DisplaySize,Connection>
Policy=Default

[Logging]
LogFile=xrdp-sesman.log
LogLevel=DEBUG
EnableSyslog=1
SyslogLevel=DEBUG

;
; Session definitions - startup command-line parameters for each session type
;

[Xorg]
; Specify the path of non-suid Xorg executable. It might differ depending
; on your distribution and version. The typical path is shown as follows:
;
; Fedora 26 or later    :  param=/usr/libexec/Xorg
; Debian 9 or later     :  param=/usr/lib/xorg/Xorg
; Ubuntu 16.04 or later :  param=/usr/lib/xorg/Xorg
; Arch Linux            :  param=/usr/bin/Xorg or param=Xorg
; CentOS 7              :  param=/usr/bin/Xorg or param=Xorg
;
param=Xorg
; Leave the rest paramaters as-is unless you understand what will happen.
param=-config
param=xrdp/xorg.conf
param=-noreset
param=-nolisten
param=tcp
param=-logfile
param=.xorgxrdp.%s.log

[X11rdp]
param=X11rdp
param=-bs
param=-nolisten
param=tcp
param=-uds

[Xvnc]
param=Xvnc
param=-bs
param=-nolisten
param=tcp
param=-localhost
param=-dpi
param=96

[Chansrv]
; drive redirection, defaults to xrdp_client if not set
FuseMountName=thinclient_drives

[SessionVariables]
PULSE_SCRIPT=/etc/xrdp/pulse/default.pa
```
xrdp.ini
```
[Globals]
; xrdp.ini file version number
ini_version=1

; fork a new process for each incoming connection
fork=true
; tcp port to listen
port=3389
; 'port' above should be connected to with vsock instead of tcp
use_vsock=false
; regulate if the listening socket use socket option tcp_nodelay
; no buffering will be performed in the TCP stack
tcp_nodelay=true
; regulate if the listening socket use socket option keepalive
; if the network connection disappear without close messages the connection will be closed
tcp_keepalive=true
#tcp_send_buffer_bytes=32768
#tcp_recv_buffer_bytes=32768

; security layer can be 'tls', 'rdp' or 'negotiate'
; for client compatible layer
security_layer=negotiate
; minimum security level allowed for client
; can be 'none', 'low', 'medium', 'high', 'fips'
crypt_level=high
; X.509 certificate and private key
; openssl req -x509 -newkey rsa:2048 -nodes -keyout key.pem -out cert.pem -days 365
certificate=
key_file=
; set SSL protocols
; can be comma separated list of 'SSLv3', 'TLSv1', 'TLSv1.1', 'TLSv1.2'
ssl_protocols=TLSv1, TLSv1.1, TLSv1.2
; set TLS cipher suites
#tls_ciphers=HIGH

; Section name to use for automatic login if the client sends username
; and password. If empty, the domain name sent by the client is used.
; If empty and no domain name is given, the first suitable section in
; this file will be used.
autorun=

allow_channels=true
allow_multimon=true
bitmap_cache=true
bitmap_compression=true
bulk_compression=true
#hidelogwindow=true
max_bpp=32
new_cursors=true
; fastpath - can be 'input', 'output', 'both', 'none'
use_fastpath=both
; when true, userid/password *must* be passed on cmd line
#require_credentials=true
; You can set the PAM error text in a gateway setup (MAX 256 chars)
#pamerrortxt=change your password according to policy at http://url

;
; colors used by windows in RGB format
;
blue=009cb5
grey=dedede
#black=000000
#dark_grey=808080
#blue=08246b
#dark_blue=08246b
#white=ffffff
#red=ff0000
#green=00ff00
#background=626c72

;
; configure login screen
;

; Login Screen Window Title
#ls_title=My Login Title

; top level window background color in RGB format
ls_top_window_bg_color=009cb5

; width and height of login screen
ls_width=350
ls_height=430

; login screen background color in RGB format
ls_bg_color=dedede

; optional background image filename (bmp format).
#ls_background_image=

; logo
; full path to bmp-file or file in shared folder
ls_logo_filename=
ls_logo_x_pos=55
ls_logo_y_pos=50

; for positioning labels such as username, password etc
ls_label_x_pos=30
ls_label_width=60

; for positioning text and combo boxes next to above labels
ls_input_x_pos=110
ls_input_width=210

; y pos for first label and combo box
ls_input_y_pos=220

; OK button
ls_btn_ok_x_pos=142
ls_btn_ok_y_pos=370
ls_btn_ok_width=85
ls_btn_ok_height=30

; Cancel button
ls_btn_cancel_x_pos=237
ls_btn_cancel_y_pos=370
ls_btn_cancel_width=85
ls_btn_cancel_height=30

[Logging]
LogFile=xrdp.log
LogLevel=DEBUG
EnableSyslog=true
SyslogLevel=DEBUG
; LogLevel and SysLogLevel could by any of: core, error, warning, info or debug

[Channels]
; Channel names not listed here will be blocked by XRDP.
; You can block any channel by setting its value to false.
; IMPORTANT! All channels are not supported in all use
; cases even if you set all values to true.
; You can override these settings on each session type
; These settings are only used if allow_channels=true
rdpdr=true
rdpsnd=true
drdynvc=true
cliprdr=true
rail=true
xrdpvr=true
tcutils=true

; for debugging xrdp, in section xrdp1, change port=-1 to this:
#port=/tmp/.xrdp/xrdp_display_10

; for debugging xrdp, add following line to section xrdp1
#chansrvport=/tmp/.xrdp/xrdp_chansrv_socket_7210


;
; Session types
;

; Some session types such as Xorg, X11rdp and Xvnc start a display server.
; Startup command-line parameters for the display server are configured
; in sesman.ini. See and configure also sesman.ini.
[Xorg]
name=Xorg
lib=libxup.so
username=ask
password=ask
ip=127.0.0.1
port=-1
code=20

[X11rdp]
name=X11rdp
lib=libxup.so
username=ask
password=ask
ip=127.0.0.1
port=-1
xserverbpp=24
code=10

[Xvnc]
name=Xvnc
lib=libvnc.so
username=ask
password=ask
ip=127.0.0.1
port=-1
#xserverbpp=24
#delay_ms=2000

[console]
name=console
lib=libvnc.so
ip=127.0.0.1
port=5900
username=na
password=ask
#delay_ms=2000

[vnc-any]
name=vnc-any
lib=libvnc.so
ip=ask
port=ask5900
username=na
password=ask
#pamusername=asksame
#pampassword=asksame
#pamsessionmng=127.0.0.1
#delay_ms=2000

[sesman-any]
name=sesman-any
lib=libvnc.so
ip=ask
port=-1
username=ask
password=ask
#delay_ms=2000

[neutrinordp-any]
name=neutrinordp-any
lib=libxrdpneutrinordp.so
ip=ask
port=ask3389
username=ask
password=ask

; You can override the common channel settings for each session type
#channel.rdpdr=true
#channel.rdpsnd=true
#channel.drdynvc=true
#channel.cliprdr=true
#channel.rail=true
#channel.xrdpvr=true
```

Xorg :10 -auth .Xauthority -config xrdp/xorg.conf -noreset -nolisten tcp

```
X.Org X Server 1.19.5
Release Date: 2017-10-12
X Protocol Version 11, Revision 0
Build Operating System: Linux 4.4.0-101-generic x86_64 Ubuntu
Current Operating System: Linux r2d2-Standard-PC-i440FX-PIIX-1996 4.13.0-39-generic #44~16.04.1-Ubuntu SMP Thu Apr 5 16:43:10 UTC 2018 x86_64
Kernel command line: BOOT_IMAGE=/boot/vmlinuz-4.13.0-39-generic root=UUID=acbf6184-6abe-4249-984e-d8dc6b080e53 ro quiet splash vt.handoff=7
Build Date: 24 November 2017  09:44:25AM
xorg-server 2:1.19.5-0ubuntu2~16.04.1 (For technical support please see http://www.ubuntu.com/support) 
Current version of pixman: 0.33.6
	Before reporting problems, check http://wiki.x.org
	to make sure that you have the latest version.
Markers: (--) probed, (**) from config file, (==) default setting,
	(++) from command line, (!!) notice, (II) informational,
	(WW) warning, (EE) error, (NI) not implemented, (??) unknown.
(==) Log file: "/home/r2d2/.local/share/xorg/Xorg.10.log", Time: Sun Apr 29 08:38:13 2018
(++) Using config file: "/etc/X11/xrdp/xorg.conf"
(==) Using system config directory "/usr/share/X11/xorg.conf.d"
(EE) xorgxrdp: module ABI major version (20) doesn't match the server's version (23)
(EE) xrdpmouse: module ABI major version (22) doesn't match the server's version (24)
(EE) xrdpkeyb: module ABI major version (22) doesn't match the server's version (24)
(EE) 
Fatal server error:
(EE) parse_vt_settings: Cannot open /dev/tty0 (Permission denied)
(EE) 
(EE) 
Please consult the The X.Org Foundation support 
	 at http://wiki.x.org
 for help. 
(EE) Please also check the log file at "/home/r2d2/.local/share/xorg/Xorg.10.log" for additional information.
(EE) 
(EE) Server terminated with error (1). Closing log file.
```

**Comment by metalefty** [comment]
@Zohman No.

```
(EE) xorgxrdp: module ABI major version (20) doesn't match the server's version (23)
(EE) xrdpmouse: module ABI major version (22) doesn't match the server's version (24)
(EE) xrdpkeyb: module ABI major version (22) doesn't match the server's version (24)
```

This is the problem. Rebuild xorgxrdp after xorg version changed.

**Comment by DS455** [comment]
I rebuilt xorgxrdp v0.2.6 the error was saved, rebuilt xorgxrdp v0.2.5 the error was again saved, overcame to xorgxrdp v0.2.6 and xrdp v0.9.5 the error is the same:
```
Xorg :10 -auth .Xauthority -config xrdp/xorg.conf -noreset -nolisten tcp

X.Org X Server 1.19.5
Release Date: 2017-10-12
X Protocol Version 11, Revision 0
Build Operating System: Linux 4.4.0-101-generic x86_64 Ubuntu
Current Operating System: Linux r2d2-Standard-PC-i440FX-PIIX-1996 4.13.0-39-generic #44~16.04.1-Ubuntu SMP Thu Apr 5 16:43:10 UTC 2018 x86_64
Kernel command line: BOOT_IMAGE=/boot/vmlinuz-4.13.0-39-generic root=UUID=acbf6184-6abe-4249-984e-d8dc6b080e53 ro quiet splash vt.handoff=7
Build Date: 24 November 2017  09:44:25AM
xorg-server 2:1.19.5-0ubuntu2~16.04.1 (For technical support please see http://www.ubuntu.com/support) 
Current version of pixman: 0.33.6
	Before reporting problems, check http://wiki.x.org
	to make sure that you have the latest version.
Markers: (--) probed, (**) from config file, (==) default setting,
	(++) from command line, (!!) notice, (II) informational,
	(WW) warning, (EE) error, (NI) not implemented, (??) unknown.
(==) Log file: "/var/log/Xorg.10.log", Time: Thu May  3 09:02:46 2018
(++) Using config file: "/etc/X11/xrdp/xorg.conf"
(==) Using system config directory "/usr/share/X11/xorg.conf.d"
(EE) xorgxrdp: module ABI major version (20) doesn't match the server's version (23)
(EE) xrdpmouse: module ABI major version (22) doesn't match the server's version (24)
(EE) xrdpkeyb: module ABI major version (22) doesn't match the server's version (24)
(II) [KMS] Kernel modesetting enabled.
resizing primary to 640x480
primary is 0x559ff7927a10
(EE) xrdpmouse: module ABI major version (22) doesn't match the server's version (24)
(EE) xrdpkeyb: module ABI major version (22) doesn't match the server's version (24)
```
Please explain in more detail how to correct this error.

**Comment by DS455** [comment]
I corrected the errors, thank you, I collected the xrdp with pulseaudio for work in the skype, but there is no sound in the skype and the microphone does not work, skype can not find device.
<a href="https://ibb.co/cD88zS"><img src="https://thumb.ibb.co/cD88zS/2018_05_03_14_53_08.png" alt="2018_05_03_14_53_08" border="0"></a>

**Comment by metalefty** [comment]
You should not report multiple issues in one issue. You should raise a new issue if you have a different issue from the original first port, generally. Regarding the issue, microphone redirection is not supported yet. See #849.
